### ISSUE TITLE
[Bug]:Multiple user sessions active after integration reloading

### ISSUE THREAD

**Issue by andreaugg94** [bvis/aegis-hass#53]
### Describe the bug

Each time the integration is reloaded, a new session for the user that was logged on is opened without closing the previous one. 
Infact in Ajax app, you can see several active sessions in the account management section.

### Steps to reproduce

Reload the integration or restarting home assistant will bring to this bug

### Expected behavior

The expected behaviour shall be the reuse of the previous session opened or to create a new one closing the previus

### Home Assistant version

2026.4.3

### Integration version

1.1.1

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

No

### Relevant logs

```text

```

**Comment by bvis** [comment]
@andreaugg94 — thanks for the report. Code review of the session lifecycle suggests the bug is plausible: we persist the session token only on the initial config flow and never refresh it after a re-login, so once the token goes stale server-side every restart re-logins and a new Ajax session is created without retiring the previous one. To confirm that's exactly what's happening on your side and not something else, I'd like to see the logs of a reload.

Could you enable debug logging for the integration and capture a reload?

**Option A — no restart needed**: HA → *Developer Tools* → *Actions* → pick `logger.set_level` → switch to YAML mode and paste:

```yaml
custom_components.aegis_ajax: debug
custom_components.aegis_ajax.api.client: debug
custom_components.aegis_ajax.api.session: debug
custom_components.aegis_ajax.coordinator: debug
```

**Option B — persistent**: add to `configuration.yaml` and restart HA:

```yaml
logger:
  default: info
  logs:
    custom_components.aegis_ajax: debug
    custom_components.aegis_ajax.api.client: debug
    custom_components.aegis_ajax.api.session: debug
    custom_components.aegis_ajax.coordinator: debug
```

Then:
1. Reload the integration once via *Settings → Devices & services → ⋮ on the Aegis entry → Reload* (no need to restart HA).
2. Check the Ajax app's *Account → Active sessions*: confirm a new entry appeared.
3. Download the log: *Settings → System → Logs → Download full log*.
4. Attach it here. The relevant lines start with `custom_components.aegis_ajax`. Feel free to redact device/space ids.

If you can also note the timestamp of the reload (e.g. "reloaded at 12:05") it helps me narrow down the events.

Also: are you on the latest stable, `v1.2.0`? Your report mentions `1.1.1`. The session-related code hasn't changed between those versions, but worth ruling out.

**Comment by bvis** [comment]
Quick update: I went ahead and shipped a fix without waiting for the logs since the pattern was clear from code review. Three things changed in **[v1.2.1-beta.4](https://github.com/bvis/aegis-hass/releases/tag/v1.2.1-beta.4)**:

1. **Session token now written back** to the config entry after every successful login — the old code only saved it during the initial setup, so a stale token on disk forced a fresh login on every restart.
2. **Auto-recovery on `UNAUTHENTICATED`** — when Ajax rejects the saved token mid-poll, the integration now re-logins, persists the new token and retries instead of failing. That breaks the "stale token → re-login → next restart same stale token" loop.
3. **`LogoutService` is called on permanent removal** — uninstalling the integration now invalidates the dangling Ajax session instead of leaving it active. Reload paths still keep the session alive on purpose, so reusing the token continues to work.

To install:
1. HACS → Integrations → *Aegis for Ajax* → ⋮ → **Redownload**
2. Enable **Show beta versions**, pick `1.2.1-beta.4`
3. Restart Home Assistant

After the restart, reload the integration once and check the *Active sessions* list in the Ajax app. With the fix the count should stop growing.

If it does keep growing, the debug logs from my earlier comment would be very helpful — there'd be a third path the code review missed.

**Comment by andreaugg94** [comment]
@bvis thanks for the fast response and sorry for my lateness.
Yesterday I've done some few tests for capturing the logs requested and I found out that the problem is not related to the integration reload but is related to the entire HA restart.

I've installed the beta you indicated, after the first HA restart the problem hasn't occoured again. Let me do some further tests and I'll come back to you soon with the logs if needed, thanks

**Comment by andreaugg94** [comment]
Hi @bvis I can confirm to you that the problem is solved. Thanks for the support!
